Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -667,8 +667,8 @@ class BrowserViewController: UIViewController {
             webViewContainerTopOffset = make.top.equalTo(readerModeBar?.snp.bottom ?? self.header.snp.bottom).constraint
 
             let findInPageHeight = (findInPageBar == nil) ? 0 : UIConstants.ToolbarHeight
-            if let toolbar = self.toolbar {
-                make.bottom.equalTo(toolbar.snp.top).offset(-findInPageHeight)
+            if let footerView = self.footer {
```

Comment:
Why does this work? `footer` will never be nil whereas `toolbar` will be when in landscape

Reply:
It's declared as optional.

Reply:
Sorry I didn't mean why do you need the `if-let`, I meant this will execute every time now and never hit the `else` below. `footer` is set and never un-set.

Reply:
Oh, so footer is constrained to bottom of view and when there is no toolbar its height becomes 0. Hence its constrained to bottom of screen. If footer will never be nil then perhaps we can remove the else case and also make it non-optional. What do you suggest?

Reply:
I'd suggest taking a closer look at why it used `toolbar` and if it really doesn't matter (like you said, its height becomes 0 anyways), then we can just remove the else clause here. In fact, we can also change the declaration of footer so its not implicitly unwrapped:

```suggestion
let footer = UIView()
```

rather than its current declaration: `var footer: UIView!` and being init'd later

Reply:
Kept it as unwrapped optional as it is initialised in viewdidload, but used without `?` as it is never set to nil.